### PR TITLE
#2701 Add conditional filterNot methods (filterNotIf/filterNotOpt) support

### PR DESCRIPTION
**Pull request by sebastianrozyczka**
Changes regarding the issue #2701

### CODE REVIEW

**Comment by sebastianrozyczka** [comment]
@nafg, I'll be grateful for the review :)

File: slick/src/main/scala/slick/lifted/Query.scala [hunk 1]
```diff
@@ -63,10 +63,22 @@ sealed abstract class Query[+E, U, C[_]] extends QueryBase[C[U]] { self =>
   def filterIf[T : CanBeQueryCondition](p: Boolean)(f: E => T): Query[E, U, C] =
     if (p) withFilter(f) else this
 
+  /** Applies the given filterNot, if the Option value is defined.
+   * If the value is None, the filter will not be part of the query. */
+  def filterNotOpt[V, T: CanBeQueryCondition](optValue: Option[V])(f: (E, V) => T): Query[E, U, C] =
+    optValue.map(v => filterHelper(e => f(e, v), node => Library.Not.typed(node.nodeType, node))
+    ).getOrElse(this)
```
Comment (nafg) [inline slick/src/main/scala/slick/lifted/Query.scala:70]:
```suggestion
    optValue
      .map(v => filterHelper(e => f(e, v), node => Library.Not.typed(node.nodeType, node)))
      .getOrElse(this)
```

**Comment by nafg** [comment]
Thanks!

**Review by nafg — APPROVED** [review]

**Comment by nafg** [comment]
@mergifyio squash

**Pull request merged**